### PR TITLE
DAT257-14-Dashboard

### PR DESCRIPTION
**Pull request by Flaskhals68**
This will add the base structure for the dashboard.
Cards have been added with place holder elements.

### CODE REVIEW

File: Client/API/ApiRequestHandler.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                return await _httpClient.GetFromJsonAsync<Country>($"{apiUrl}/Country/GetCountryOfTheDay");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw new Exception($"Api call failed, {e}");
+            }
+        }
+
+        public static async Task<Country> FetchCountry(int id)
```
Comment (filipkauffeldt) [inline Client/API/ApiRequestHandler.cs]: It should be a string with the country ISO code to the database. The endpoint is not up to date with that yet, but should also be changed

File: Client/API/Country.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+namespace API
+{
+    public class Country
+    {
+        public required string Code { get; init; }
+        public required string Name { get; init; }
+        public string? Continent { get; init; }
+        public string? Description { get; init; }
+        public Collection<Data>? Data { get; init; }
+
```
Comment (filipkauffeldt) [inline Client/API/Country.cs:13]: Remove new line

File: Client/Layout/MainLayout.razor [hunk 3]
```diff
@@ -5,8 +5,11 @@
     </div>
 
     <main>
-        <div class="top-row px-4">
+@*         <div class="top-row px-4">
```
Comment (filipkauffeldt) [inline Client/Layout/MainLayout.razor]: Remove unused

**Comment by filipkauffeldt** [comment]
Should the APIRequestHandler maybe be injected as a dependency in the program.cs file and have a http client injected? I think that is best practice, and also allows for testing the methods by mocking the http client

**Review by filipkauffeldt — COMMENTED** [review]
Looks good, but should the api request handler not be tested?

**Review by jakobschy — DISMISSED** [review]
Looks good

**Review by erbergli — APPROVED** [review]
LGTM!!!!

**Review by Huffla — APPROVED** [review]
TYes

**Pull request merged**